packages/EasyApiPlatform/src/Filter/AdvancedSearchFilter.php

```diff
@@ -0,0 +1,485 @@
+                    $filterParameterNames[] = $filterParameterName . '[]';
+                }
+
+                foreach ($filterParameterNames as $filterParameterName) {
+                    $description[$filterParameterName] = [
+                        'property' => $propertyName,
+                        'type' => $typeOfField,
+                        'required' => false,
+                        'strategy' => $strategy,
+                        'is_collection' => str_ends_with($filterParameterName, '[]'),
+                    ];
+                }
+            } elseif ($metadata->hasAssociation($field)) {
+                $filterParameterNames = [
+                    $filterParameterName,
+                    $filterParameterName . '[]',
+                ];
+
+                foreach ($filterParameterNames as $filterParameterName) {
+                    $description[$filterParameterName] = [
+                        'property' => $propertyName,
+                        'type' => 'string',
+                        'required' => false,
+                        'strategy' => self::STRATEGY_EXACT,
+                        'is_collection' => str_ends_with($filterParameterName, '[]'),
+                    ];
+                }
+            }
+        }
+
+        return $description;
+    }
+
+    /**
+     * Adds where clause according to the strategy.
+     *
+     * @throws InvalidArgumentException If strategy does not exist
+     */
+    protected function addWhereByStrategy(
+        string $strategy,
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $alias,
+        string $field,
+        mixed $values,
+        bool $caseSensitive
+    ): void {
+        if (is_array($values) === false) {
+            $values = [$values];
+        }
+
+        $wrapCase = $this->createWrapCase($caseSensitive);
+        $valueParameter = ':' . $queryNameGenerator->generateParameterName($field);
+        $aliasedField = sprintf('%s.%s', $alias, $field);
+
+        if ($strategy === '' || $strategy === self::STRATEGY_EXACT) {
+            if (count($values) === 1) {
+                $queryBuilder
+                    ->andWhere($queryBuilder->expr()->eq($wrapCase($aliasedField), $wrapCase($valueParameter)))
+                    ->setParameter($valueParameter, $values[0]);
+
+                return;
+            }
+
+            $queryBuilder
+                ->andWhere($queryBuilder->expr()->in($wrapCase($aliasedField), $valueParameter))
+                ->setParameter($valueParameter, $caseSensitive ? $values : array_map('strtolower', $values));
+
+            return;
+        }
+
+        $ors = [];
+        $parameters = [];
+        foreach ($values as $key => $value) {
+            $keyValueParameter = sprintf('%s_%s', $valueParameter, $key);
+            $parameters[] = [$caseSensitive ? $value : strtolower($value), $keyValueParameter];
```

Comment:
It doesn't look the same as:
```
$parameters[$caseSensitive ? $value : strtolower($value)] = $keyValueParameter;
```

Reply:
Oh, I see that's because of `array_walk` in the end. But why don't you keep the original structure, as in SearchFilter?

Reply:
Would you please recheck with `\ApiPlatform\Doctrine\Orm\Filter\SearchFilter`?

Reply:
I see, thank you.